### DISCUSSION TITLE
posts/cpp-vtable-and-object-memory-layout-clang-example/

### DISCUSSION THREAD

**Discussion by giscus** [pluveto/blog-comments#4] [Announcements]
# posts/cpp-vtable-and-object-memory-layout-clang-example/

https://www.less-bug.com/posts/cpp-vtable-and-object-memory-layout-clang-example/

**Comment by Diana010717** [comment]
深度好文👍👍👍总算讲清楚了

**Reply by Diana010717** [reply]
-24是距离vcalloffset的距离😂

**Comment by Diana010717** [comment]
vcalloffset的-24是距离this指针的距离，-32也是因为是第二个函数它的vcalloffset是在前一个上面的数据，对应vcalloffset是反的，vbaseoffset一直是-24，都是RRTI、offsettotop上面的那个数

**Reply by Diana010717** [reply]
-24、-32都是虚表中的距离

**Reply by pluveto** [reply]
感谢指正！
